internal/docstore/query.go

```diff
@@ -144,3 +153,26 @@ func (it *DocumentIterator) Stop() {
 	it.err = io.EOF
 	it.iter.Stop()
 }
+
+// A QueryPlan describes how a query will be executed.
```

Comment:
I'm a bit nervous about putting too much structure here given that we have an n of only 1. WDYT about just returning a string instead? Drivers can put whatever description they want, but it's intended for a human to read it.

Reply:
Done.